neptune/observation.go

```diff
@@ -2,13 +2,59 @@ package neptune
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"strings"
 
+	"github.com/ONSdigital/dp-graph/neptune/query"
 	"github.com/ONSdigital/dp-graph/observation"
 	"github.com/ONSdigital/dp-observation-importer/models"
 )
 
+// ErrEmptyFilter is returned if the provided filter is empty.
+var ErrEmptyFilter = errors.New("filter is empty")
+
 func (n *NeptuneDB) StreamCSVRows(ctx context.Context, filter *observation.Filter, limit *int) (observation.StreamRowReader, error) {
-	return nil, nil
+	if filter == nil {
+		return nil, ErrEmptyFilter
+	}
+
+	q := fmt.Sprintf(query.GetInstanceHeader, filter.InstanceID)
+
+	q += buildObservationsQuery(filter)
+	q += query.GetObservationSelectRowPart
+
+	if limit != nil {
+		q += fmt.Sprintf(query.LimitPart, *limit)
+	}
+
+	return n.Pool.OpenCursorCtx(ctx, q, nil, nil)
+}
+
+func buildObservationsQuery(f *observation.Filter) string {
+	if f.IsEmpty() {
```

Comment:
Here we react to a filter that can be judged to be *Empty* by using a query that does not filter. Conversely, in StreamCsvRows, we judge a nil filter to be an error, but describe that error as ErrEmptyFilter. Perhaps we should either rename ErrEmptyFilter to ErrNilFilter, or treat an *Empty* filter as an error here?

Reply:
Yeah the name of the error is misleading. There is a meaningful difference between a nil filter and an empty filter - where an empty filter is requesting no subset, and thus the entirety, of an instance. A nil filter doesn't even provide us an instance ID to be able to run a query with. Will look at changing the error name